src/module/utils/model-collection.mjs

```diff
@@ -38,50 +67,86 @@ export default class ModelCollection extends foundry.utils.Collection {
   /* -------------------------------------------------- */
 
   /**
-   * A set of the un-initialized pseudo-documents.
-   * Stored safely for debugging purposes.
-   * @type {Set<object>}
+   * The set of invalid document ids.
+   * @param {Set<string>}
    */
-  #invalid = new Set();
+  invalidDocumentIds = new Set();
 
   /* -------------------------------------------------- */
   /*  Methods                                           */
   /* -------------------------------------------------- */
 
   /**
-   * Fetch an array of data models of a certain type.
-   * @param {string} type     The subtype of the data models.
-   * @returns {Model[]}   The data models of this type.
+   * The subtypes sorted by type.
+   * @type {Record<string, Model[]>}
    */
-  getByType(type) {
-    return Array.from(this.#types.get(type) ?? []).map(key => this.get(key));
+  get documentsByType() {
```

Comment:
The previous iteration updated `#types` with each addition or removal. Like with embedded collections, this getter now instead handles the assignment of `#types` (now `#documentsByType`) and other methods are responsible for `null`'ing it so it can be updated.